sonar-plugin-api/src/main/java/org/sonar/api/measures/Metric.java

```diff
@@ -699,7 +699,11 @@ public Builder setDomain(String d) {
      *
      * @param f the formula
      * @return the builder
+     *
+     * @deprecated since 5.2, it's no more possible to define a formula on a metric
+     * TODO add a link to the new API to declare formulas
      */
+    @Deprecated
     public Builder setFormula(Formula f) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)